test/e2e/framework/ssh/ssh.go

```diff
@@ -133,13 +134,41 @@ func NodeSSHHosts(c clientset.Interface) ([]string, error) {
 			len(hosts), len(nodelist.Items), nodelist)
 	}
 
-	sshHosts := make([]string, 0, len(hosts))
-	for _, h := range hosts {
-		sshHosts = append(sshHosts, net.JoinHostPort(h, SSHPort))
+	lenHosts := len(hosts)
+	wg := &sync.WaitGroup{}
+	wg.Add(lenHosts)
+	sshHosts := make([]string, 0, lenHosts)
+	sshHostsLock := &sync.Mutex{}
+
+	for _, host := range hosts {
+		go func(host string) {
+			if canConnect(host) {
```

Comment:
Nit: `defer wg.Done()` at the start ensures that it is called even when someone decides to add a `return` somewhere below later.

Reply:
Thanks, fixed as suggested :pray: